### PR TITLE
ddns-updater: init at 2.6.0

### PR DESCRIPTION
**Pull request by delliottxyz**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/dd/ddns-updater/package.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "-w"
+  ];
+
+  meta = with lib; {
+    description = "Container to update DNS records periodically with WebUI for many DNS providers";
+    homepage = "https://github.com/qdm12/ddns-updater";
+    license = licenses.mit;
+    maintainers = with maintainers; [delliott];
+    mainProgram = "updater";
+    platforms = platforms.linux;
```
Comment (nevivurn) [inline pkgs/by-name/dd/ddns-updater/package.nix:30]: The most recent release (2.6.0) provides darwin binaries, might be worth looking into.

File: pkgs/by-name/dd/ddns-updater/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "-s"
+    "-w"
+  ];
+
+  meta = with lib; {
+    description = "Container to update DNS records periodically with WebUI for many DNS providers";
+    homepage = "https://github.com/qdm12/ddns-updater";
+    license = licenses.mit;
+    maintainers = with maintainers; [delliott];
+    mainProgram = "updater";
```
Comment (nevivurn) [inline pkgs/by-name/dd/ddns-updater/package.nix:29]: Upstream calls the binary `ddns-updater`. Maybe rename it in `postInstall`.

**Comment by nevivurn** [comment]
Oh, and the commit should be renamed as `ddns-updater: init at 2.5.0` (remove the v)

**Comment by delliottxyz** [comment]
@nevivurn Thank you for the review. I think the issues raised have been adequately resolved?

**Review by nevivurn — APPROVED** [review]
Result of `nixpkgs-review pr 284822` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ddns-updater</li>
  </ul>
</details>

File: pkgs/by-name/dd/ddns-updater/package.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    mv $out/bin/updater $out/bin/ddns-updater
+  '';
+
+  meta = with lib; {
+    description = "Container to update DNS records periodically with WebUI for many DNS providers";
+    homepage = "https://github.com/qdm12/ddns-updater";
+    license = licenses.mit;
+    maintainers = with maintainers; [ delliott ];
+    mainProgram = "ddns-updater";
+    platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/by-name/dd/ddns-updater/package.nix:34]:
```suggestion
```
that's the default for buildGoModule

File: pkgs/by-name/dd/ddns-updater/package.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+  vendorHash = "sha256-Ibrv0m3Tz/5JbkHYmiJ9Ijo37fjHc7TP100K7ZTwO8I=";
+
+  ldflags = [
+    "-s"
+    "-w"
+  ];
+
+  postInstall = ''
+    mv $out/bin/updater $out/bin/ddns-updater
```
Comment (SuperSandro2000) [inline pkgs/by-name/dd/ddns-updater/package.nix:25]: We should probably set ``subPackages = [ "cmd/updater" ];`` to not compile something else by accident

**Comment by delliottxyz** [comment]
Thank you for the review @SuperSandro2000. I believe the issues raised have been resolved?

**Pull request merged**